changelog.d/pr-10817

```diff
@@ -0,0 +1,15 @@
+synopsis: Fix PATH changes not triggering REPL reconfiguration
+packages: cabal-install
+prs: #10817
+issues: #2015
+
+When modifying your PATH environment variable between Cabal commands, REPL
+sessions will now correctly use the updated environment. Previously, if you
+changed your PATH after initial configuration, Cabal would continue using the
+old PATH settings in REPL sessions without warning.
+
+With this fix, Cabal properly detects PATH changes and reconfigures the REPL
+accordingly, ensuring tools and libraries in your current PATH are available.
```

Comment:
This text, including the title, feels a little repetitive. I suggest dropping the first sentence in the description and make it into one paragraph:

```suggestion
Previously, if you changed your PATH after initial configuration, Cabal would
continue using the old PATH settings in REPL sessions without warning.
With this fix, Cabal properly detects PATH changes and reconfigures the REPL
accordingly, ensuring tools and libraries in your current PATH are available.
```
Again, consider that it appears right after the title ("synopsis" field). And that the former third sentence basically repeated the former first one that I suggest to drop.